### PR TITLE
[[ TRAFODION 1800 ]] Changes for Backup/Restore

### PR DESCRIPTION
**Pull request by amandamoran**

### CODE REVIEW

**Comment by DaveBirdsall** [comment]
Looks good to me.

File: install/installer/traf_add_sudoAccess [hunk 1]
```diff
@@ -42,4 +53,31 @@ if [[ $all_node_count -gt "1" ]]; then
...
+fi
+
+}
+
+
+#Main
+echo "***INFO: Creating $TRAF_USER sudo access file"
+echo
+
+if [[ "$ENABLE_HA" == "Y" ]]; then
```
Comment (arvind-narain) [inline install/installer/traf_add_sudoAccess]: ENABLE_HA gets set to true or false - should this check be for "true" ?

File: install/installer/traf_config_check [hunk 2]
```diff
@@ -487,6 +487,18 @@ if [[ "$SUSE_LINUX" != "true" ]]; then
...
 
+function checkBackupUser {
+
+if [ -z $BACKUP_USER ]; then
+   echo "***ERROR: BACKUP_USER variable not set in config file."
+   echo "***WARNING: BACKUP_USER will be set to trafodion."
+   sudo chmod 777 $TRAF_CONFIG
+   sudo "export BACKUP_USER=\"trafodion\"" >> $TRAF_CONFIG
+   sudo chmod 777 $TRAF_CONFIG
+fi
```
Comment (arvind-narain) [inline install/installer/traf_config_check]: Should "source $TRAF_CONFIG" be done here after setting BACKUP_USER ?

File: install/installer/traf_cloudera_mods98 [hunk 3]
```diff
@@ -158,6 +158,21 @@ if [ $? != 0 ]; then
 fi
 sudo su $HDFS_USER --command "$HADOOP_BIN_PATH/hadoop fs -chown -R $TRAF_USER:trafodion /lobs"
 
+
+#Create Backup directory 
+sudo su $HDFS_USER --command "$HADOOP_BIN_PATH/hadoop fs -mkdir /trafodion_backups" 2> $HOME/traf_temp_output
+if [ $? != 0 ]; then
+   # ok if directory already exists
+   dir_exists=$(grep "File exists" $HOME/traf_temp_output | wc -l)
```
Comment (svarnau) [inline install/installer/traf_cloudera_mods98]: I think this code could be simplified by using -p option to the -mkdir command. Then it should not give error for directory exists.

**Comment by DaveBirdsall** [comment]
Where are we at with this change? Are further changes planned?

**Comment by amandamoran** [comment]
I was hoping to address the issues on another check in, if thats okay.

On Wed, Feb 17, 2016 at 8:34 AM, DaveBirdsall <notifications@github.com>
wrote:

> Where are we at with this change? Are further changes planned?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-trafodion/pull/302#issuecomment-185286019>
> .
>



-- 
Thanks,

Amanda Moran

**Comment by DaveBirdsall** [comment]
Of course!

**Comment by amandamoran** [comment]
Sorry, I think I phrased that funny.

I meant another pull request. So maybe we could merge these changes for
now?

On Wed, Feb 17, 2016 at 10:14 AM, DaveBirdsall <notifications@github.com>
wrote:

> Of course!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-trafodion/pull/302#issuecomment-185330733>
> .
>



-- 
Thanks,

Amanda Moran

**Comment by DaveBirdsall** [comment]
Go for it!

**Pull request merged**